File: DigitalTwin/Client/FSR.DigitalTwin.ROS2/Scripts/fsr/digital_twin/abc/workspace.py
from abc import ABC, abstractmethod
from enum import Enum

class WorkspaceKind(Enum):
    DIGITAL = 0
    VIRTUAL = 1
    PHYSICAL = 2

class Workspace(ABC):
    @abstractmethod
    def _get_kind(self):
        pass

    @property
    def kind(self):
        return self._get_kind()

File: DigitalTwin/Client/FSR.DigitalTwin.Unity/Assets/Scripts/DigitalTwin/Workspace/Workspace.cs
using System.Collections;
using System.Collections.Generic;
using FSR.DigitalTwinLayer.GRPC.Lib;
using UnityEngine;

namespace FSR.DigitalTwin.Unity.Workspace {
public abstract class Workspace : MonoBehaviour
{
    public enum EKind {
        DIGITAL, VIRTUAL, PHYSICAL
    }

    public abstract EKind Kind { get; }
}

}


